src/Voron/Impl/Journal/JournalReader.cs

```diff
@@ -161,28 +174,38 @@ private bool TryReadAndValidateHeader(StorageEnvironmentOptions options, out Tra
             current = (TransactionHeader*)
                 (_journalPager.AcquirePagePointer(this, pageNumber) + positionInsidePage);
 
+            // due to the reuse of journals we no longer can assume we have zeros in the end of the journal
+            // we might have there random garbage or old transactions we can ignore, so we have the following scenarios:
+            // * TxId <= current Id      ::  we can ignore old transaction of the reused journal and continue
+            // * TxId == current Id + 1  ::  valid, but if hash is invalid - we can assume corruption
+            // * TxId >  current Id + 1  ::  if hash is invalid we can ignore reused/random, but if hash valid then we might missed TXs 
+
             if (current->HeaderMarker != Constants.TransactionHeaderMarker)
             {
                 // not a transaction page, 
 
-                // if the header marker is zero, we are probably in the area at the end of the log file, and have no additional log records
+                // if the header marker is zero or garbage, we are probably in the area at the end of the log file, and have no additional log records
                 // to read from it. This can happen if the next transaction was too big to fit in the current log file. We stop reading
-                // this log file and move to the next one. 
+                // this log file and move to the next one, or it might have happened becuase of reuse of journal file 
 
-                RequireHeaderUpdate = current->HeaderMarker != 0;
-                if (RequireHeaderUpdate)
-                {
-                    options.InvokeRecoveryError(this, "Transaction " + current->TransactionId + " header marker was set to garbage value, file is probably corrupted", null);
-                }
```

Comment:
Didn't we loose the validation check and raising recovery error here? I don't see `options.InvokeRecoveryError` to be moved somewhere.